Makefile

```diff
@@ -476,14 +476,18 @@ create-tag: require-version
 	@git tag -a "$$VERSION" -f --annotate -m"Tagged $$VERSION" && \
 	git push --tags -f
 
+# This target requires a file called testpackages.txt that lists all packages to
+# test, one package per line, with no trailing newline on the last package.
 test-and-cover: $(RESOURCES_DOT_GO)
 	@echo "mode: count" > profile.cov && \
 	source setenv.bash && \
 	if [ -f envvars.bash ]; then \
 		source envvars.bash; \
 	fi && \
-	for pkg in $$(cat testpackages.txt); do \
-		go test -race -v -tags="headless" -covermode=atomic -coverprofile=profile_tmp.cov $$pkg || exit 1; \
+	TP=$$(cat testpackages.txt) && \
+	CP=$$(echo -n $$TP | tr ' ', ',') && \
+	for pkg in $$TP; do \
+		go test -race -v -tags="headless" -covermode=atomic -coverpkg "$$CP" -coverprofile=profile_tmp.cov $$pkg || exit 1; \
```

Comment:
So what was happening before is that we weren't specifying the coverpkg flag.  Without that flag, cover only counts coverage for code in the package that's being tested.  So if the test for that package happens to exercise a lot of other packages, we don't get credit for it.  By specifying coverpkg, we get credit for other packages, so what I'm doing is specifying all testpackages as -coverpkg whenever we test any of those testpackages.  This way, when we run the integration test in flashlight/app, we get credit for all the code in our other packages that this test also exercises.